### ISSUE TITLE
DiffusionNFT Exhibits High Variance between Runs

### ISSUE THREAD

**Issue by Morefre** [NVlabs/DiffusionNFT#47]
Hi authors,
I am currently running DiffusionNFT for Geneval using the default hyperparameters (the only difference is I am running on 4 gpus). I noticed that across different random seeds, the model behavior seems to exhibit relatively high variance.

Specifically, as shown in the first figure below, the average norm of x_0 differs significantly from run to run and sometimes gradually increases during training. This corresponds to noticeably brighter generated images (see the second figure). Interestingly, however, the average reward appears to remain relatively stable across runs (see the third figure).

I was wondering whether this behavior is expected, or if it might indicate that something is wrong on my side.

Additionally, I would be interested to know approximately how many epochs were required to reach the 0.98 Geneval score reported in Figure 1 of the paper.

Thank you very much for your time, and congratulations on the ICLR oral acceptance!

<img width="736" height="303" alt="Image" src="https://github.com/user-attachments/assets/311d9123-58a3-4117-9408-ae7e2c64ed60" />

<img width="2122" height="219" alt="Image" src="https://github.com/user-attachments/assets/57800495-233a-4028-a1bf-34ea568863e6" />

<img width="933" height="267" alt="Image" src="https://github.com/user-attachments/assets/9ee22127-826d-44e4-b6bd-5e472f36391f" />

**Comment by jaylee2000** [comment]
@Morefre I've observed similar phenomena in DDPO, Flow-GRPO, DanceGRPO.
Specifically, between runs with the same effective batch size & hyperparameters, auxiliary metrics (e.g., max grad norm) showed high variance. Consequently, the images produced by checkpoints varied a lot as well.
Similar to what you observed, the average reward showed relatively small variance between runs.

I haven't tested a lot with DiffusionNFT - I'd love to learn whether it was an error on your side or not.
